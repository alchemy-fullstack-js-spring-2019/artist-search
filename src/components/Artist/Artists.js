import React from 'react';
import PropTypes from 'prop-types';
import Artist from './Artist';

function Artists({ artistArray }) {
  const artistList = artistArray.map(artist => {
    return (
      <li key={ artist.id }>
        <Artist id={ artist.id } name={ artist.name } />
      </li>
    );
  });

  return (
    <ul>
      { artistList }
    </ul>
  );
}

Artists.propTypes = {
  artistArray: PropTypes.array.isRequired
};

export default Artists;
