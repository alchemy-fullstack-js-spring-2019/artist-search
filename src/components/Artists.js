import React from 'react';
import PropTypes from 'prop-types';
import Artist from './Artist';
import { Link } from 'react-router-dom';

function Artists({ artistArray }){
  const artistList = artistArray.map((artist, i) => {
    const artistId = `/${artist.name}/${artist.id}`;
    return (
      
      <li key={i}>
        <Link to={artistId}><Artist artist={artist}/></Link>
      </li>
    );
  });

  return (
    <ul>
      {artistList}
    </ul>
  );
}

Artists.propTypes = {
  artistArray: PropTypes.array.isRequired
};

export default Artists;
