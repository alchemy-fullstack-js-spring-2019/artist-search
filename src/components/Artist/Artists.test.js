import React from 'react';
import { shallow } from 'enzyme';
import Artists from './Artists';

describe('Artists component', () => {  
  it('renders the Artists', () => {
    const fakeData = [
      {
        id: '123',
        name: 'Shakira'
      },
      {
        id: '321',
        name: 'Beyonce'
      }
    ];
    
    const wrapper = shallow(<Artists artistArray={ fakeData }/>);
    expect(wrapper).toMatchSnapshot();
  });
});
