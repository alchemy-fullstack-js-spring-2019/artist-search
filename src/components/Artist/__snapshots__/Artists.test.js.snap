// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artists component renders the Artists 1`] = `
<ul>
  <li
    key="123"
  >
    <Artist
      id="123"
      name="Shakira"
    />
  </li>
  <li
    key="321"
  >
    <Artist
      id="321"
      name="Beyonce"
    />
  </li>
</ul>
`;
